roboslack-api/src/main/java/com/palantir/roboslack/api/time/FormatToken.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.roboslack.api.time;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
+import com.palantir.roboslack.api.markdown.StringDecorator;
+import com.palantir.roboslack.api.markdown.ValueDecorator;
+import java.util.Optional;
+import java.util.stream.Stream;
+import javax.annotation.CheckForNull;
+
+/**
+ * Enumerated types for describing date and time formats in the way Slack expects.
+ * All format descriptions are provided using the {@link java.time.format.DateTimeFormatter} symbol designations.
+ *
+ * @see <a href="https://api.slack.com/docs/message-formatting">Message Formatting</a>
+ * @see java.time.format.DateTimeFormatter
+ * @since 1.0.0
+ */
+@JsonSerialize(using = ToStringSerializer.class)
+public enum FormatToken {
+    /**
+     * {@code MMMM dd(th/st/nd/rd), yyyy} <br/>
+     * (ie. {@code February 18th, 2014}).
+     */
+    DATE("MMMMM dd, yyyy"),
+    /**
+     * {@code yyyy-mm-dd} <br/>
+     * (ie. {@code 2014-02-18})
+     */
+    DATE_NUM("yyyy-mm-dd"),
+    /**
+     * {@code MMM dd, yyyy} <br/>
+     * (ie. {@code Feb 18, 2014})
+     */
+    DATE_SHORT("MMM dd, yyyy"),
+    /**
+     * {@code eeee, MMMMM dd(th/st/nd/rd), yyyy}
+     * (ie. {@code Tuesday, February 18th, 2014})
+     */
+    DATE_LONG("eeee, MMMM dd, yyyy"),
+    /**
+     * Same as {@link #DATE} but uses "yesterday", "today", or "tomorrow" where appropriate.
+     */
+    DATE_PRETTY("MMMM dd, yyyy"), // Delegate to just using DATE's pattern
+    /**
+     * Same as {@link #DATE_SHORT} but uses "yesterday", "today", or "tomorrow" where appropriate.
+     */
+    DATE_SHORT_PRETTY("MMM dd, yyyy"), // Delegate to just using DATE_SHORT's pattern
```

Comment:
Each pattern is different, and they aren't being used in more than one place. I'd be inclined to leave as-is.